docs/config.md

```diff
@@ -122,18 +124,304 @@ model_names: <comma-delimited-string|list>
 
 # List of GPU UUIDs to be used for the profiling. Use 'all' to profile all the GPUs visible by CUDA."
 [ gpus: <string|comma-delimited-list-string> | default: 'all' ]
+
+# List of constraints placed on the config search results.
+[ constrains: <constraint> ]
+
+# List of objectives that user wants to sort the results by it.
+[ objectives: <objective|list> ]
+
+# Maximum concurrency used for the automatic config search.
+[ run_config_search_max_concurrency: <int> | default: 1024 ]
+
+# Maximum instance count used for the automatic config search.
+[ run_config_search_max_instance_count: <int> | default: 5 ]
+
+# Maximum instance count used for the automatic config search.
+[ run_config_search_max_preferred_batch_size: <int> | default: 16 ]
+
+# Disables automatic config search
+[ run_config_search_disable: <bool> | default: false ]
 ```
 
-If you save configuration file in `config.yml`, you can provide
-the path to the Model Analyzer config using the 
-`-f` flag.
+## `<constraint>`
+A constraint, specifies the bounds that determine a successful run. There are
+three constraints allowed:
```

Comment:
Why are we limiting to just three constraints? Shouldn't we be able to constrain any measurable output?

Reply:
@aramesh7 Can we support other constraints too? I can add them to the config.

Reply:
Can we make this a table? Something like this?

| Option Name          | Units       |  Constraint  |   Description                                                      |
| :--------------------- |  :-------:  |  :----------:  |  :---------------------------------------------------------- |
| `perf_throughput`     | inf / sec  |      min       | Specify minimum desired throughput.                           |
| `perf_latency`            | ms          |       max     | Specify maximum tolerable latency or latency budget. |
| `gpu_used_memory` | MB         |       max     | Specify maximum GPU memory used by model.           |

Reply:
We do support constraints on any metric we measure